test_scripts/Image_template/commonImageTemplate.lua

```diff
@@ -0,0 +1,190 @@
+
+--[[ Module ]]
+local m = actions
+
+--[[ Local variables ]]
+local putFileParams = {
+	requestParams = {
+	    syncFileName = 'icon.png',
+	    fileType = "GRAPHIC_PNG",
+	    persistentFile = false,
+	    systemFile = false
+	},
+	filePath = "files/icon.png"
+}
+
+--[[ Functions ]]
+local function addCommandParams()
+	local requestParams = {
+		cmdID = 11,
+		menuParams = {
+			position = 0,
+			menuName ="Commandpositive"
+		},
+		cmdIcon = {
+			value ="icon.png",
+			imageType ="DYNAMIC"
+		}
+	}
+	local responseUiParams = {
+		cmdID = requestParams.cmdID,
+		cmdIcon = commonFunctions:cloneTable(requestParams.cmdIcon),
+		menuParams = requestParams.menuParams
+	}
+	responseUiParams.cmdIcon.value = m.getPathToFileInStorage("icon.png")
+	local params = {
+		requestParams = requestParams,
+		responseUiParams = responseUiParams,
+	}
+	 return params
+end
+
+local function alertParams()
+	local requestParams = {
+	alertText1 = "alertText1",
+	alertText2 = "alertText2",
+	softButtons = {
+		{
+			type = "BOTH",
+			text = "Close",
+			image = {
+				value = "icon.png",
+				imageType = "DYNAMIC",
+			},
+			isHighlighted = true,
+			softButtonID = 3,
+			systemAction = "DEFAULT_ACTION",
+		}
+	}
+}
+
+local responseUiParamsAlert = {
+	alertStrings = {
+		{
+			fieldName = requestParams.alertText1,
+			fieldText = requestParams.alertText1
+		},
+		{
+			fieldName = requestParams.alertText2,
+			fieldText = requestParams.alertText2
+		}
+	},
+		alertType = "UI",
+		softButtons = commonFunctions:cloneTable(requestParams.softButtons)
+	}
+	responseUiParamsAlert.softButtons[1].image.value = m.getPathToFileInStorage("icon.png")
+	local params = {
+		requestParams = requestParams,
+		responseUiParams = responseUiParamsAlert,
+	}
+	return params
+end
+
+function m.getPathToFileInStorage(fileName)
+  return commonPreconditions:GetPathToSDL() .. "storage/"
+  .. config["application1"].registerAppInterfaceParams.appID .. "_"
+  .. utils.getDeviceMAC() .. "/" .. fileName
+end
+
+function m.putFile(pParams)
+  if not pParams then pParams = putFileParams end
+  local mobileSession = m.getMobileSession();
+  local cid = mobileSession:SendRPC("PutFile", pParams.requestParams, pParams.filePath)
+
+  mobileSession:ExpectResponse(cid, { success = true, resultCode = "SUCCESS"})
+end
+
+function m.addCommand(pIsTemplate, pParams)
+	if not pParams then pParams = addCommandParams() end
+	if
+		pIsTemplate == true or
+	 	pIsTemplate == false then
+		pParams.requestParams.cmdIcon.isTemplate = pIsTemplate
+		pParams.responseUiParams.cmdIcon.isTemplate = pIsTemplate
+	end
+	local mobSession = m.getMobileSession()
+	local hmiConnection = m.getHMIConnection()
+	local cid = mobSession:SendRPC("AddCommand", pParams.requestParams)
+	pParams.responseUiParams.appID = m.getHMIAppId()
+	EXPECT_HMICALL("UI.AddCommand", pParams.responseUiParams)
+	:Do(function(_,data)
+		hmiConnection:SendResponse(data.id, data.method, "SUCCESS", {})
+	end)
+	:ValidIf(function(_, data)
+		if
```

Comment:
@GetmanetsIrina These conditions won't catch the following case:
1. pIsTemplate = nil
2. data.params.cmdIcon.isTemplate = false

Result of ValidIf will be true, however isTemplate wasn't sent

The same issue is for ```Alert``` function